### PR TITLE
Make `paging-testing` multiplatform by moving everything to `common*`

### PR DESCRIPTION
**Pull request by veyndan**
Test: ./gradlew paging:paging-testing:allTests

### CODE REVIEW

File: paging/paging-testing/build.gradle [hunk 1]
```diff
@@ -33,20 +39,57 @@ androidXMultiplatform {
...
+                target.compilations["main"].defaultSourceSet {
+                    def konanTargetFamily = target.konanTarget.family
+                    if (konanTargetFamily == Family.OSX || konanTargetFamily == Family.IOS) {
+                        dependsOn(darwinMain)
+                    } else if (konanTargetFamily == Family.LINUX) {
+                        dependsOn(linuxMain)
+                    } else {
+                        throw new GradleException("unknown native target ${target}")
+                    }
+                }
```
Comment (veyndan) [inline paging/paging-testing/build.gradle:92]: This was mostly a copy-paste from `paging-common`'s `build.gradle` file.

**Comment by veyndan** [comment]
CI should be fixed by https://android-review.googlesource.com/c/platform/frameworks/support/+/2746657. I'll rebase this PR once that has been merged.

**Review by claraf3 — APPROVED** [review]
Thanks for this!

**Review by ianhanniballake — APPROVED** [review]

**Comment by dlam** [comment]
Copybara is having trouble mirroring this because of the conflict.

**Comment by claraf3** [comment]
Infra recently upgraded their build and changed `androidTest` references to `androidUnitTest`. Can you please try rebasing?

**Comment by veyndan** [comment]
Rebased!

**Pull request closed without merging**